src/control/server/harness.go

```diff
@@ -149,12 +147,12 @@ func (h *IOServerHarness) AwaitStorageReady(ctx context.Context, skipMissingSupe
 	h.RLock()
 	defer h.RUnlock()
 
-	if h.started {
+	if h.IsStarted() {
 		return errors.New("can't wait for storage: harness already started")
 	}
 
 	h.log.Info("Waiting for I/O server instance storage to be ready...")
-	for _, instance := range h.instances {
+	for _, instance := range h.Instances() {
```

Comment:
While this might work OK, it's asking for trouble. Go's read locks aren't meant to be used for recursive locking: https://golang.org/pkg/sync/#RWMutex.RLock